### ISSUE TITLE
2.0 + WebpackOptionsValidationError

### ISSUE THREAD

**Issue by blitzd** [zeit/next.js#1646]
Every version since 2.0 has given me the following error on Windows 10:

```
{ WebpackOptionsValidationError: Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration.context: The provided value "c:\\Users\\blitzd\\Source\\Repos\\hello-next" is not an absolute path!
 - configuration.module.rules[4].include: The provided value "c:\\Users\\blitzd\\Source\\Repos\\hello-next\\node_modules\\next\\dist\\pages" is not an absolute path!
 - configuration.output.path: The provided value "c:\\Users\\blitzd\\Source\\Repos\\hello-next\\.next" is not an absolute path!
    at webpack (c:\Users\blitzd\Source\Repos\hello-next\node_modules\next\node_modules\webpack\lib\webpack.js:19:9)
    at _callee2$ (c:\Users\blitzd\Source\Repos\hello-next\node_modules\next\dist\server\build\webpack.js:529:69)
    at tryCatch (c:\Users\blitzd\Source\Repos\hello-next\node_modules\regenerator-runtime\runtime.js:64:40)
    at Generator.invoke [as _invoke] (c:\Users\blitzd\Source\Repos\hello-next\node_modules\regenerator-runtime\runtime.js:299:22)
    at Generator.prototype.(anonymous function) [as next] (c:\Users\blitzd\Source\Repos\hello-next\node_modules\regenerator-runtime\runtime.js:116:21)
    at step (c:\Users\blitzd\Source\Repos\hello-next\node_modules\babel-runtime\helpers\asyncToGenerator.js:17:30)
    at c:\Users\blitzd\Source\Repos\hello-next\node_modules\babel-runtime\helpers\asyncToGenerator.js:35:14
    at F (c:\Users\blitzd\Source\Repos\hello-next\node_modules\babel-runtime\node_modules\core-js\library\modules\_export.js:35:28)
    at c:\Users\blitzd\Source\Repos\hello-next\node_modules\babel-runtime\helpers\asyncToGenerator.js:14:12
    at createCompiler (c:\Users\blitzd\Source\Repos\hello-next\node_modules\next\dist\server\build\webpack.js:540:17)
  name: 'WebpackOptionsValidationError',
  message: 'Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.\n - configuration.context: The provided value "c:\\\\Users\\\\blitzd\\\\Source\\\\Repos\\\\hello-next" is not an absolute path!\n - configuration.module.rules[4].include: The provided value "c:\\\\Users\\\\blitzd\\\\Source\\\\Repos\\\\hello-next\\\\node_modules\\\\next\\\\dist\\\\pages" is not an absolute path!\n - configuration.output.path: The provided value "c:\\\\Users\\\\blitzd\\\\Source\\\\Repos\\\\hello-next\\\\.next" is not an absolute path!',
  validationErrors:
   [ { keyword: 'absolutePath',
       params: [Object],
       message: 'The provided value "c:\\\\Users\\\\blitzd\\\\Source\\\\Repos\\\\hello-next" is not an absolute path!',
       parentSchema: [Object],
       dataPath: '.context',
       schemaPath: '#/properties/context/absolutePath',
       schema: true,
       data: 'c:\\Users\\blitzd\\Source\\Repos\\hello-next' },
     { keyword: 'absolutePath',
       params: [Object],
       message: 'The provided value "c:\\\\Users\\\\blitzd\\\\Source\\\\Repos\\\\hello-next\\\\node_modules\\\\next\\\\dist\\\\pages" is not an absolute path!',
       parentSchema: [Object],
       dataPath: '.module.rules[4].include',
       schemaPath: '#/properties/include/allOf/1/absolutePath',
       schema: true,
       data: 'c:\\Users\\blitzd\\Source\\Repos\\hello-next\\node_modules\\next\\dist\\pages' },
     { keyword: 'absolutePath',
       params: [Object],
       message: 'The provided value "c:\\\\Users\\\\blitzd\\\\Source\\\\Repos\\\\hello-next\\\\.next" is not an absolute path!',
       parentSchema: [Object],
       dataPath: '.output.path',
       schemaPath: '#/definitions/output/properties/path/absolutePath',
       schema: true,
       data: 'c:\\Users\\blitzd\\Source\\Repos\\hello-next\\.next' } ] }
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

This is when running the 'yarn run dev' (or npm run dev, same basic error) step outlined in the tutorial here:

https://learnnextjs.com/basics/getting-started/setup

**Comment by blitzd** [comment]
This is related to an issue in webpack here:

https://github.com/webpack/webpack/issues/4530

The gist is that it's because of a lowercase drive letter in the path. The fix is to start your shell or command prompt with an uppercase drive letter, or do something along the lines of ```cd .. && cd C:\<path>```.

This could probably be resolved in the webpack setup for next. Judging by the webpack issue they don't intend of changing that behaviour.

I have not run into this with cmd32.exe or powershell.exe, but the default Anaconda Command Prompt does have a lowercase drive letter for the 'starts in' path that it gets set up with, and that's where I ran across this.

**Comment by arunoda** [comment]
I assume this will be fixed with newer webpack release. And we'll ship that with the next version.

**Comment by blitzd** [comment]
@arunoda it looks like it should be resolved in 2.3.3 here:

https://github.com/webpack/webpack/commit/56d09ab65c3c340f2d2dfad258799901c45b92a3

**Comment by arunoda** [comment]
Okay. We are [using it](https://github.com/zeit/next.js/blob/master/package.json#L92).
Get our latest release: https://github.com/zeit/next.js/releases
